src/many-macros/src/lib.rs

```diff
@@ -166,52 +397,48 @@ fn many_module_impl(attr: &TokenStream, item: TokenStream) -> Result<TokenStream
     let info_name = format!("{}Info", struct_name);
     let info_ident = Ident::new(&info_name, attr.span());
 
-    let endpoints: Result<Vec<_>, (String, Span)> = trait_
+    let endpoints: Vec<Endpoint> = tr
         .items
         .iter()
-        .filter_map(|item| match item {
-            TraitItem::Method(m) => Some(m),
-            _ => None,
+        .filter_map(|item| {
+            if let TraitItem::Method(m) = item {
+                Some(Endpoint::new(m))
+            } else {
+                None
+            }
         })
-        .map(|item| Endpoint::new(&item.sig))
-        .collect();
-    let endpoints = endpoints.map_err(|(msg, span)| syn::Error::new(span, msg))?;
-    let ns = namespace.clone();
+        .collect::<syn::Result<_>>()?;
+
+    let trait_ = {
+        let attributes = tr.attrs.iter();
+        quote! {
+            #(#attributes)*
+            #vis trait #trait_ident: Send {
```

Comment:
Is `Send` mandatory? If so, could we add a check?

Reply:
Removed it and replaced with super-traits from the original. Good catch!